src/core/thread/dua_manager.cpp

```diff
@@ -75,23 +76,98 @@ void DuaManager::UpdateDomainUnicastAddress(BackboneRouter::Leader::DomainPrefix
+    mDomainUnicastAddress.GetAddress().SetPrefix(prefix->mPrefix.mFields.m8, prefix->mLength);
 
-    if (Get<Utils::Slaac>().GenerateIid(mDomainUnicastAddress, NULL, 0, &mDadCounter) == OT_ERROR_NONE)
+    // Apply cached DUA Interface Identifier manually specified.
+    if (!mFixedDuaInterfaceIdentifier.IsUnspecified())
     {
-        Get<ThreadNetif>().AddUnicastAddress(mDomainUnicastAddress);
+        mDomainUnicastAddress.GetAddress().SetIid(mFixedDuaInterfaceIdentifier);
     }
     else
     {
-        mDomainUnicastAddress.GetAddress().Clear();
-        otLogWarnCore("Failed to generate valid DUA");
+        SuccessOrExit(GenerateDomainUnicastAddressIid());
     }
 
+    Get<ThreadNetif>().AddUnicastAddress(mDomainUnicastAddress);
+
 exit:
     return;
 }
 
+otError DuaManager::GenerateDomainUnicastAddressIid(void)
+{
+    otError error;
+
+    if ((error = Get<Utils::Slaac>().GenerateIid(mDomainUnicastAddress, NULL, 0, &mDadCounter)) == OT_ERROR_NONE)
+    {
+        otLogInfoCore("Generated DUA: %s", mDomainUnicastAddress.GetAddress().ToString().AsCString());
+    }
+    else
+    {
+        otLogWarnCore("Failed to generate valid DUA");
+    }
+
+    return error;
+}
+
+otError DuaManager::SetFixedDuaInterfaceIdentifier(const Ip6::InterfaceIdentifier *aIid)
+{
+    otError error = OT_ERROR_NONE;
+
+    // Clear the fixed Interface Identifier manually specified if there was.
+    if (aIid == NULL)
+    {
+        // Nothing to clear.
+        VerifyOrExit(!mFixedDuaInterfaceIdentifier.IsUnspecified(), OT_NOOP);
+
+        if (GetDomainUnicastAddress().HasIid(mFixedDuaInterfaceIdentifier) &&
+            Get<ThreadNetif>().IsUnicastAddress(GetDomainUnicastAddress()))
+        {
+            Get<ThreadNetif>().RemoveUnicastAddress(mDomainUnicastAddress);
+
+            if (GenerateDomainUnicastAddressIid() == OT_ERROR_NONE)
+            {
+                Get<ThreadNetif>().AddUnicastAddress(mDomainUnicastAddress);
+            }
+        }
+
+        otLogInfoCore("Cleared DUA IID: %s", mFixedDuaInterfaceIdentifier.ToString().AsCString());
+        mFixedDuaInterfaceIdentifier.Clear();
+    }
+    else
+    {
+        VerifyOrExit(!aIid->IsReserved(), error = OT_ERROR_INVALID_ARGS);
+        VerifyOrExit(mFixedDuaInterfaceIdentifier.IsUnspecified() || mFixedDuaInterfaceIdentifier != *aIid, OT_NOOP);
+
+        mFixedDuaInterfaceIdentifier = *aIid;
+        otLogInfoCore("Set DUA IID: %s", mFixedDuaInterfaceIdentifier.ToString().AsCString());
+
+        if (Get<ThreadNetif>().IsUnicastAddress(GetDomainUnicastAddress()))
+        {
+            Get<ThreadNetif>().RemoveUnicastAddress(mDomainUnicastAddress);
+            mDomainUnicastAddress.GetAddress().SetIid(mFixedDuaInterfaceIdentifier);
+            Get<ThreadNetif>().AddUnicastAddress(mDomainUnicastAddress);
+        }
+    }
+
+exit:
+    return error;
+}
+
+const Ip6::InterfaceIdentifier *DuaManager::GetFixedDuaInterfaceIdentifier(void) const
```

Comment:
Similar suggestion here:
- Adding a new API, `IsFixedDuaInterfaceIdentifierSet()` to indicate if it set.
- And getter returning a reference (and only used when it it set?). 

Thoughts?